src/App.tsx

```diff
@@ -1,24 +1,98 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Footer } from './components/footer/Footer';
+import { Todo } from './types/Todo';
+import { deleteTodo, getTodos } from './api/todos';
+import { Header } from './components/header/Header';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './types/TodoFelter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+const USER_ID = 11960;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentFilter, setCurrentFilter]
+    = useState<TodoFilter>(TodoFilter.All);
+  const [errorText, setErrorText] = useState<Error>(Error.Default);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorText(Error.CantLoad));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (currentFilter) {
+      case TodoFilter.Active:
+        return todos.filter(todo => !todo.completed);
+      case TodoFilter.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, currentFilter]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleNewTodo = (todo: Todo) => {
+    setTodos(current => [...current, todo]);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(todoId)
+      .catch(() => setErrorText(Error.Delete))
+      .finally(() => {
+        setTodos(current => current.filter(todo => todo.id !== todoId));
+      });
+  };
+
+  const handleDeleteCompleted = async (todosId: number[]) => {
+    try {
+      for (let i = 0; i < todosId.length; i += 1) {
+        deleteTodo(todosId[i]);
+      }
+    } catch (error) {
+      setErrorText(Error.Delete);
+    } finally {
+      setTodos(current => current.filter(todo => !todosId.includes(todo.id)));
+    }
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          newTodo={handleNewTodo}
+          setTempTodo={setTempTodo}
+          idUser={USER_ID}
+          setErrorMessege={setErrorText}
+        />
+
+        {todos.length > 0 && (
```

Comment:
```suggestion
        {!!todos.length && (
```
If a mentor points out a mistake, it means that you need to fix ALL occurrences of the mistake, not only in the place where the comment was left.